azure/services/subnets/subnets.go

```diff
@@ -99,6 +99,12 @@ func (s *Service) Reconcile(ctx context.Context) error {
 				}
 			}
 
+			if subnetSpec.NatGatewayName != "" {
```

Comment:
Nat Gateways are reconciled before subnets. When reconciling subnets, if a nat gateway was specified, then it's safe to use its Azure ID.